test/extensions/filters/network/common/fuzz/uber_writefilter.cc

```diff
@@ -0,0 +1,121 @@
+#include "test/extensions/filters/network/common/fuzz/uber_writefilter.h"
+
+#include "common/config/utility.h"
+#include "common/config/version_converter.h"
+
+using testing::_;
+using testing::Return;
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+void UberWriteFilterFuzzer::reset() {
+  // Reset some changes made by current filter on some mock objects.
+
+  // Close the connection to make sure the filter's callback is set to nullptr.
+  write_filter_callbacks_->connection_.raiseEvent(Network::ConnectionEvent::LocalClose);
+  // Clear the filter's raw pointer stored inside the connection_ and reset the connection_'s state.
+  write_filter_callbacks_->connection_.callbacks_.clear();
+  write_filter_callbacks_->connection_.bytes_sent_callbacks_.clear();
+  write_filter_callbacks_->connection_.state_ = Network::Connection::State::Open;
+  // Clear the pointers inside the mock_dispatcher
+  Event::MockDispatcher& mock_dispatcher =
+      dynamic_cast<Event::MockDispatcher&>(write_filter_callbacks_->connection_.dispatcher_);
+  mock_dispatcher.to_delete_.clear();
+  write_filter_.reset();
+}
+
+void UberWriteFilterFuzzer::fuzzerSetup() {
+  // Setup process when this fuzzer object is constructed.
+  // For a static fuzzer, this will only be executed once.
+
+  // Get the pointer of write_filter when the write_filter is being added to connection_.
+  write_filter_callbacks_ = std::make_shared<NiceMock<Network::MockWriteFilterCallbacks>>();
+  read_filter_callbacks_ = std::make_shared<NiceMock<Network::MockReadFilterCallbacks>>();
+  ON_CALL(write_filter_callbacks_->connection_, addWriteFilter(_))
+      .WillByDefault(Invoke([&](Network::WriteFilterSharedPtr write_filter) -> void {
+        write_filter->initializeWriteFilterCallbacks(*write_filter_callbacks_);
+        write_filter_ = write_filter;
+      }));
+  ON_CALL(write_filter_callbacks_->connection_, addFilter(_))
+      .WillByDefault(Invoke([&](Network::FilterSharedPtr filter) -> void {
+        filter->initializeReadFilterCallbacks(*read_filter_callbacks_);
+        filter->initializeWriteFilterCallbacks(*write_filter_callbacks_);
+        write_filter_ = filter;
+      }));
+  factory_context_.prepareSimulatedSystemTime();
+  // Set featureEnabled for mongo_proxy
+  ON_CALL(factory_context_.runtime_loader_.snapshot_, featureEnabled("mongo.proxy_enabled", 100))
+      .WillByDefault(Return(true));
+  ON_CALL(factory_context_.runtime_loader_.snapshot_,
+          featureEnabled("mongo.connection_logging_enabled", 100))
+      .WillByDefault(Return(true));
+  ON_CALL(factory_context_.runtime_loader_.snapshot_, featureEnabled("mongo.logging_enabled", 100))
+      .WillByDefault(Return(true));
+  // Set featureEnabled for thrift_proxy
```

Comment:
Above this line, you set up state for the mongo_proxy filter.

Below this line, you set up state for the thrift filter.

Put a newline between these sections, so that it is easy to see where configuration for mongo ends and thrift begins.

Reply:
> Above this line, you set up state for the mongo_proxy filter.
> 
> Below this line, you set up state for the thrift filter.
> 
> Put a newline between these sections, so that it is easy to see where configuration for mongo ends and thrift begins.

Added, thanks!